fitbenchmarking/parsing/hogben_parser.py

```diff
@@ -0,0 +1,72 @@
+"""
+This file implements a parser for HOGBEN problem sets.
+"""
+import os
+import typing
+import pickle
+import numpy as np
+
+from fitbenchmarking.parsing.fitbenchmark_parser import FitbenchmarkParser
+
+
+class HogbenParser(FitbenchmarkParser):
+    """
+    Parser for a Hogben problem definition file.
+    """
+
+    def _create_function(self) -> typing.Callable:
+        """
+        Process the reflectometry model into a callable.
+
+        Expected function format:
+        function='function=model.pkl'
+
+        :return: A callable function
+        :rtype: callable
+        """
+
+        pf = self._parsed_func[0]
+        model = os.path.join(os.path.dirname(self._filename),
+                             "Models", pf['function'])
+        with open(model, 'rb') as f:
+            refnx_model = pickle.load(f)
+
+        # pylint: disable=attribute-defined-outside-init
+        self._equation = pf['function'].split('.')[0].replace('_', ' ')
+
+        varying_params = refnx_model.parameters.varying_parameters()
+
+        # remove spaces and hyphens from parameter names
+        p_names = [" ".join(p.name.replace("-", "").split()).replace(" ", "_")
+                   for p in list(varying_params)]
```

Comment:
Do params have strange whitespace (that's handled by split) or is this equivalent?
```suggestion
        p_names = [p.name.replace("-", "").replace(" ", "_")
                   for p in list(varying_params)]
```

Reply:
The param names include a hyphen with whitespace on either side, so to avoid having a double underscore in the middle of the parameter name (due to double whitespace after removing the hyphen) I added in the split and join step to remove the extra whitespace